src/deluge/model/song/song.cpp

```diff
@@ -5781,6 +5792,29 @@ int32_t Song::convertSyncLevelFromInternalValueToFileValue(int32_t internalValue
 	return fileValue;
 }
 
+const char* Song::getSongFullPath() {
+	String fullPath;
+	fullPath.concatenate(&dirPath);
+	fullPath.concatenate("/");
+	fullPath.concatenate(&name);
+	return fullPath.get();
```

Comment:
This is unsafe, `fullPath`'s storage will be deallocated at the end of the function and the returned pointer will therefore be dangling. You should return the string object itself.